File: app/lib/models/room_model.dart
class Room {
  late final String building;
  late final int capacity;
  late final String roomNumber;
  late final String roomName;
  late final String type;

  Room({
    required this.building,
    required this.capacity,
    required this.roomNumber,
    required this.type,
    required this.roomName
  });

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      building: json['building'],
      capacity: json['capacity'],
      roomNumber: json['room_number'],
      type: json['type'],
      roomName: json['room_name'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'building': building,
      'capacity': capacity,
      'room_number': roomNumber,
      'type': type,
      'room_name': roomName,
    };
  }
}

File: admin/src/models/roomModel.js
class Room {
    constructor({ room_id, building, capacity, room_number, type }) {
        this.room_id = room_id;
        this.building = building;
        this.capacity = capacity;
        this.room_number = room_number;
        this.type = type;
    }
}

export default Room;
